pkg/tuf/client.go

```diff
@@ -313,10 +313,34 @@ func NewFromEnv(_ context.Context) (*TUF, error) {
 	}
 
 	// Initializes a new TUF object from the local cache or defaults.
-	return initializeTUF(mirror, nil, getEmbedded(), false)
+	return initializeTUFOnce(mirror, nil, getEmbedded(), false)
 }
 
+// Initialize creates a new TUF client, if one has not already been initialized,
+// and writes the cache to disk if requested. Thread safe.
```

Comment:
this doesn't seem threadsafe from a quick glance and from the PR description.

Reply:
Also probably worth while to note in the comments that this is the singleton instance. Or make it somehow clear that it does not necessairly return a new TUF client, but could return a previously created one. Reasoning being that it would then be easier to understand why this is different from InitializeAlways.